### PR TITLE
Pivot table

### PR DESCRIPTION
**Pull request by Mikhail-iontsev**
Closes https://github.com/OHDSI/Ares/issues/115

Added pivot table to the data quality results component. The component is created to be reusable.

The attribute box items can be dragged to change the headers layout.

### CODE REVIEW

File: src/interface/views/release/DataQualityResults.vue [hunk 1]
```diff
@@ -406,15 +418,20 @@ import deriveResults from "@/services/derive-results";
...
     return {
       chooseHeaderMenu: false,
       chooseFilter: false,
       versions: [],
+      currentTag: "",
       dataLoaded: false,
       dqResults: null,
+      selection: [],
+      rawData: [],
+      pivotTableHeaders: [],
```
Comment (fdefalco) [inline src/interface/views/release/DataQualityResults.vue:434]: Is pivotTableHeaders used anywhere?

**Review by fdefalco — CHANGES_REQUESTED** [review]
Is there a way to filter the data fields available in the list to drag/drop? The columns CHECK_DESCRIPTION, CONCEPT_ID, EXECUTION_TIME, NUM_DENOMINATOR_ROWS, NUM_VIOLATED_ROWS, PCT_VIOLATED_ROWS, QUERY_TEXT,  THRESHOLD_VALUE, UNIT_CONCEPT_ID, WARNING, _row, and checkId should be removed.

If you click on the word 'Table' in the top left cell you can change the viewer type, can this be disabled?

Font family varies across the component (verdana) and Ares (Roboto).  Can we standardize on Roboto.  I prefer no bold text in the pivot view results column.

**Comment by Mikhail-iontsev** [comment]
> Is there a way to filter the data fields available in the list to drag/drop? The columns CHECK_DESCRIPTION, CONCEPT_ID, EXECUTION_TIME, NUM_DENOMINATOR_ROWS, NUM_VIOLATED_ROWS, PCT_VIOLATED_ROWS, QUERY_TEXT, THRESHOLD_VALUE, UNIT_CONCEPT_ID, WARNING, _row, and checkId should be removed.
> 
> If you click on the word 'Table' in the top left cell you can change the viewer type, can this be disabled?
> 
> Font family varies across the component (verdana) and Ares (Roboto). Can we standardize on Roboto. I prefer no bold text in the pivot view results column.

1. Yes, we can. 
2.  We can disable the type change button but along with all the other interface elements except for the table itself. In this case we will need write separate row and column table constructors.
3. So the table headers can stay bold?

**Comment by fdefalco** [comment]
3. I don't think they need to be bold. They can be consistent with the overview tab where the background color of the header row is differentiated but the font is not bold.

**Comment by Mikhail-iontsev** [comment]
> 3. I don't think they need to be bold. They can be consistent with the overview tab where the background color of the header row is differentiated but the font is not bold.

So, there are two ways to disable the renderer type button: 

1. Just plain disable it, but the cell containing it remains: 
Screenshot from 2022-04-23 00-11-33
2. Create custom controls: 
Screenshot from 2022-04-23 00-15-17

Which one do you prefer?

**Comment by fdefalco** [comment]
The custom controls seem to have a much more integrated appearance.  I prefer them.

File: src/interface/views/release/DataQualityResults.vue (hunk 1 above)
Comment (Mikhail-iontsev) [reply src/interface/views/release/DataQualityResults.vue]: Forgot to delete. It was created along with a few functions for a table using v-data-table component.

**Comment by fdefalco** [comment]
Looks great.  Only issue is that the custom control font size is 16px, can you change to 14px?

**Pull request merged**